### PR TITLE
feat(admin-ui): install applications status handler

### PR DESCRIPTION
**Pull request by frdomovic**
# feat(admin-ui): install applications status handler
 
## Summary:
Added popup status modal for installing applications. It can show success status and error status. Success status on continue navigates you to applications list and resets the state, while error status closes the modal and leaves you on the select package and release page to try again.

Fixes # [C20-54](https://calimero.atlassian.net/browse/C20-54)

### CODE REVIEW

File: node-ui/src/hooks/useAdminClient.js [hunk 1]
```diff
@@ -2,11 +2,28 @@ import axios from "axios";
...
+    try {
+      const response = await axios.post("/admin-api/install-application", {
+        application: selectedPackage,
+        version: selectedVersion,
+      });
+      if (response.status === 200) {
+        return { data: response.data };
+      } else {
+        return {
+          error: { code: 500, message: "Failed to install application" },
```
Comment (MatejVukosav) [inline node-ui/src/hooks/useAdminClient.js]: Use error returned from server

File: node-ui/src/hooks/useAdminClient.js [hunk 2]
```diff
@@ -2,11 +2,28 @@ import axios from "axios";
...
+        };
+      }
+    } catch (error) {
+      if (error.response) {
```
Comment (MatejVukosav) [inline node-ui/src/hooks/useAdminClient.js:12]: Can silently ignore error if `error.response `does not exist.

File: node-ui/src/pages/Applications.jsx [hunk 3]
```diff
@@ -45,29 +53,64 @@ export default function Applications() {
...
+            setShowInstallApplications={setShowInstallApplications}
+            showStatusModal={showStatusModal}
+            closeModal={closeStatusModal}
+            installationStatus={installationStatus}
           />
         ) : (
           <ApplicationsTable
             applications={applications}
-            install={() => setSwitchInstall(true)}
+            install={() => setShowInstallApplications(true)}
```
Comment (MatejVukosav) [inline node-ui/src/pages/Applications.jsx]: I think a different approach is needed here. As this is in the form of tabs, the tabs work on a base where a single item is selected.  To enable tabs to grow it is pretty messy to maintain semaphore states with on and off. A better way is to use a selected item where you define what should be visible in the tab and not which item is on and off. Saying that remove `setShowInstallApplications` and define an enum with possibilities and iterate over options.

Comment (frdomovic) [reply node-ui/src/pages/Applications.jsx]: Updated with JS version of enum

File: node-ui/src/hooks/useAdminClient.js (hunk 1 above)
Comment (frdomovic) [reply node-ui/src/hooks/useAdminClient.js]: updated

File: node-ui/src/hooks/useAdminClient.js (hunk 2 above)
Comment (frdomovic) [reply node-ui/src/hooks/useAdminClient.js:12]: ?

Comment (MatejVukosav) [reply node-ui/src/hooks/useAdminClient.js:12]: If an error does not have a response field, an error will not be shown to the user right? We should have a generic error in that case indicating that something is wrong and we don't have more details about the problem so the user knows that his action has not finished successfully.

Comment (frdomovic) [reply node-ui/src/hooks/useAdminClient.js:12]: Updated , will be displayed as:
title: Failed to install application
sub: Try again later.

**Pull request merged**